### PR TITLE
r23 converters for MID

### PR DESCRIPTION
**Pull request by dstocco**

### CODE REVIEW

File: aliroot/r23/dataformatconverter/ConvertDE.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+namespace mid
+{
+/// Set of utilities allowing to pass from the old Run2 formats to the new one
+/// and viceversa.
+/// These utilities are meant for testing, allowing one to compare
+/// with the AliRoot input/output on existing data.
+/// It is of course doomed to disappear with time.
+struct ConvertDE
+{
+  static int convertFromLegacyDeId(int detElemId);
```
Comment (aphecetche) [inline aliroot/r23/dataformatconverter/ConvertDE.h]: Why wrapping this single function into a struct ?
Let it be free function simply (alo.mid is local enough IMO).

File: aliroot/r23/trackrefsconverter/mid-convert-trackrefs.cxx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    std::ofstream outFile(outFilename.c_str(), std::ios::binary);
+
+    for (auto input : inputfiles)
+    {
+        std::cout << "Converting file " << input << "\n";
+        trackRefsToMID(input.c_str(), outFile);
+    }
+    outFile.close();
+
+    // // REMEMBER TO CUT
```
Comment (aphecetche) [inline aliroot/r23/trackrefsconverter/mid-convert-trackrefs.cxx]: Remove that commented part simply ?

File: o2/mid/readers/BoostReader.h [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  bool read(std::vector<T>& vec, int nObjects)
+  {
+    T obj;
+    while (read(obj)) {
+      vec.emplace_back(obj);
+    }
+    return true;
+  }
+
+//   template <class T>
```
Comment (aphecetche) [inline o2/mid/readers/BoostReader.h]: remove that commented part ?

**Review by aphecetche — CHANGES_REQUESTED** [review]

File: aliroot/r23/dataformatconverter/ConvertDE.h (hunk 1 above)
Comment (dstocco) [reply aliroot/r23/dataformatconverter/ConvertDE.h]: Let function be..

File: aliroot/r23/trackrefsconverter/mid-convert-trackrefs.cxx (hunk 2 above)
Comment (dstocco) [reply aliroot/r23/trackrefsconverter/mid-convert-trackrefs.cxx]: Removed

File: o2/mid/readers/BoostReader.h (hunk 3 above)
Comment (dstocco) [reply o2/mid/readers/BoostReader.h]: Removed

**Review by aphecetche — APPROVED** [review]

**Pull request merged**